### ISSUE TITLE
limit(3**n*3**(-n - 1)*(n + 1)**2/n**2, n, oo) is wrong

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#10610]
```
In [41]: limit(3**n*3**(-n - 1)*(n + 1)**2/n**2, n, oo)
Out[41]: 1
```

From https://stackoverflow.com/questions/35428617/sympy-fails-limit-test. If you expand the expression first the terms cancel and it gets the answer right. The answer should be 1/3.

**Comment by jksuom** [comment]
It seems that the test ``ok`` referred to in [this comment](https://github.com/sympy/sympy/blob/master/sympy/series/limits.py#L148) is too liberal. It accepts all expressions that contain a polynomial both in the numerator and the denominator. In this case there are also dominating exponentials that essentially determine the limit, but they are ignored by the test.

**Comment by govindlahoti** [comment]
I am looking into it

**Comment by govindlahoti** [comment]
I am looking into this

**Comment by wrat** [comment]
if Everything all right(all test cases has been passed ) then why this pull request is merged yet???

**Comment by govindlahoti** [comment]
@wrat There are still test cases that can fail

**Comment by vassbu** [comment]
Yes,
here is one:
>>> x = symbols('x')
>>> n = symbols('n', integer=True)
>>> a = (x-1)**n/(2**n*(n+1))
>>> a1 = a.subs(n, n+1)
>>> L = limit(a1/a, n, oo)
>>> L
1                                      # Bug i SymPy!!
>>> L = limit(apart(a1/a), n, oo)      # workaround
>>> L
x/2 - 1/2

**Comment by viditjain1** [comment]
is this issue solved?

**Comment by govindlahoti** [comment]
@viditjain1  No.
I am not working on this bug anymore. Sorry

**Comment by jksuom** [comment]
Fixed by #12121.
